### PR TITLE
feat(agroal): Adds a configuration option to enable datasource metrics

### LINKED ISSUE #2820: Agroal datasource metrics enablement
**Description**
It should be made possible to enable agroal metrics gathering via configuration file.

it can still be done programatically but both deployment and runtime config classes simply ignore the metrics point.

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #2820

### CODE REVIEW

File: extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java [hunk 1]
```diff
@@ -91,4 +91,9 @@
     @ConfigItem
     public Optional<TransactionIsolationLevel> transactionIsolationLevel;
 
+    /**
+     * The datasource metrics enablement
```
Comment (gsmet) [inline extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java]:
```suggestion
     * Enable datasource metrics collection.
```

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 2]
```diff
@@ -277,6 +277,10 @@ AgroalDataSource dataSource2;
 |io.quarkus.agroal.runtime.TransactionIsolationLevel (enum)
 |none, read-committed, read-uncommitted, repeatable-read, serializable
 
+|quarkus.datasource.enable-metrics
+|Boolean
+|True or False
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: Mmh, I think we should make it `true` or `false` all lowercase.

Same for the other booleans.

**Review by gsmet — COMMENTED** [review]
I suggested two minor changes. I knew I would have your attention by adding the `good-first-issue` tag but didn't think you would be that fast ;).

File: docs/src/main/asciidoc/datasource-guide.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/datasource-guide.adoc]: I'll make the other modification in this same commit.

**Comment by machi1990** [comment]
> I suggested two minor changes. I knew I would have your attention by adding the `good-first-issue` tag but didn't think you would be that fast ;).

Ha ha yes, I had some time and knew where to look at :-)

**Review by gsmet — APPROVED** [review]
Looks good to me.

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**